crates/gossip/src/gossip.rs

```diff
@@ -251,6 +251,18 @@ impl K2Gossip {
             target_peer_url
         );
 
+        // Before we send the initiate message, check whether the target has already
+        // initiated with us. If they have, we can just skip initiating.
+        if self
+            .accepted_round_states
+            .read()
+            .await
+            .contains_key(&target_peer_url)
+        {
+            tracing::info!("initiate_gossip: already accepted");
+            return Ok(false);
+        }
+
```

Comment:
I've reverted that =D

Reply:
Haha, that's why it's showing up on this diff. I was confused by that because my rebase went through without a conflict. Cool, let me revert here!